network/src/noise/handshake.rs

```diff
@@ -179,44 +187,54 @@ impl NoiseUpgrader {
     /// Perform an outbound protocol upgrade on this connection.
     ///
     /// This runs the "client" side of the Noise IK handshake to establish a
-    /// secure Noise stream and exchange static public keys. In mutual auth
-    /// scenarios, we will also include an anti replay attack counter in the
+    /// secure Noise stream and send its static public key to the server.
+    /// In mutual auth scenarios, we will also include an anti replay attack counter in the
     /// Noise handshake payload. Currently this counter is always a millisecond-
     /// granularity unix epoch timestamp.
-    pub async fn upgrade_outbound<TSocket>(
+    pub async fn upgrade_outbound_with_noise<TSocket>(
         &self,
         mut socket: TSocket,
         remote_public_key: x25519::PublicKey,
     ) -> io::Result<NoiseStream<TSocket>>
     where
         TSocket: AsyncRead + AsyncWrite + Unpin,
     {
-        // send a payload of the current timestamp (in milliseconds)
+        // craft prologue = self_peer_id | expected_public_key
+        let mut prologue = [0; PeerId::LENGTH + x25519::PUBLIC_KEY_SIZE];
+        prologue[..PeerId::LENGTH].copy_from_slice(self.self_peer_id.as_ref());
+        prologue[PeerId::LENGTH..].copy_from_slice(remote_public_key.as_slice());
+
+        // craft 8-byte payload as current timestamp (in milliseconds)
         let payload = {
             let now: u64 = time::SystemTime::now()
                 .duration_since(time::UNIX_EPOCH)
                 .expect("system clock should work")
-                .as_millis() as u64;
+                .as_millis() as u64; // (PAYLOAD_SIZE)
+
             // e.g. [157, 126, 253, 97, 114, 1, 0, 0]
             let now = now.to_le_bytes().to_vec();
```

Comment:
does not need `to_vec` can just return `to_le_bytes` which should just give us an array